src/App.js

```diff
@@ -1,7 +1,24 @@
+import MyMap from './components/MyMap'
+import UserLocationText from './components/UserLocationText'
+import CoffeeShopsText from './components/CoffeeShopsText'
+import useLocation from './hooks/useLocation'
+import useCoffeeShops from './hooks/useCoffeeShops'
+
 function App() {
+  const userLocationState = useLocation()
+  const coffeeShopsState = useCoffeeShops()
```

Comment:
This looks much cleaner. As a side note, I would argue against extracting EVERYTHING into a new hook. `useLocation()` makes sense because it is a piece of state that is self contained and changes on its own, The coffee shops on the other hand belong to the API and you need to request them, it would probably make more sense if you just abstracted the Data Layer in some sort of API Service. But this is good enough for now. Once again, much better!